New Thermostat example (#3675)
icerpc/icerpc-csharp: examples/Thermostat/README.md

# Thermostat

> This larger example presents an application with a client (think app running on your phone) that interacts with a device

Reviewer: ```suggestion
This larger example presents an application with a client (ex. an app running on your phone) that interacts with a device
```

I think saying think three times in a row feels a bit repetitive. Breaking this up with ex. might be good.

> This larger example presents an application with a client (think app running on your phone) that interacts with a device

Reviewer: ```suggestion
This more extensive example presents an application with a client (think app running on your phone) that interacts with a device
```

I think large almost have a bit of a negative context, I think more extensive is a better word choice.

Author: Removed larger.

> (think thermostat in your home) via an intermediary server (think server deployed on a public cloud).

Reviewer: ```suggestion
(ex. a server deployed on a public cloud) via an intermediary server (think server deployed on a public cloud).
```

Author: Reworked text.

## Interactions

```mermaid
---
title: Typical deployment
---
flowchart LR
    subgraph cloud [Public cloud]
        Server
    end
    subgraph home [Home network]
        Device
    end
    subgraph shop [Coffee shop]
        Client
    end
    Client -- port 4062 --> Server

    %% Put device on the right hand size
    Server ~~~ Device
    Device -- port 10000 --> Server
```

The arrows on the diagram indicate the client and server sides of the connection, not the flow of requests and
responses.

Server implements two Slice interfaces using two separate services:
 - `Thermostat`, the client-facing interface
 - `ThermoHome`, the device-facing interface

Server also calls a service on the device that implements Slice interface `ThermoControl`.

## Build and run

You can build the client, server and device with:

``` shell
dotnet build
```

First start the Server program:

```shell
cd Server
dotnet run
```

In a separate window, start the Device program:

```shell
cd Device
dotnet run
```

Then in another window, start the Client program:

```shell
cd Client
dotnet run
```

> You can start multiple instances of the Client program. When you start Client without argument, it monitors the device

Reviewer 2: Maybe this? I'm not sure if you were trying to say something else though.
```suggestion
You can start multiple instances of the Client program. When you start Client without any arguments, it monitors the device
```

Author: Reworked text
